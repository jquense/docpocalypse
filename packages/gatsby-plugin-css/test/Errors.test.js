/* eslint-disable jest/no-disabled-tests */

const { webpack } = require('./helpers');

describe('Errors', () => {
  xtest('Validation Error', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          sourceMap: 1,
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      console.log(source);
      // eslint-disable-next-line
      const error = () => eval(source);

      expect(error).toThrow();

      try {
        error();
      } catch (err) {
        const message = err.message.split('\n\n').slice(1, -1).join('\n\n');

        expect(message).toMatchSnapshot();
      }
    });
  });

  test('Syntax Error', () => {
    const config = {
      loader: {
        test: /\.css$/,
        options: {
          parser: 'sugarss',
        },
      },
    };

    return webpack('css/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      // eslint-disable-next-line
      const error = () => eval(source);

      expect(error).toThrow();

      try {
        error();
      } catch (err) {
        const message = err.message.split('\n').slice(1).join('\n');

        expect(message).toMatchSnapshot();
      }
    });
  });
});
