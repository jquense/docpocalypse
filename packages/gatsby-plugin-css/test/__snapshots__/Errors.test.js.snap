// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Syntax Error 1`] = `
"SyntaxError

(1:3) Unexpected separator in property

[31m[1m>[22m[39m[90m 1 | [39ma [33m{[39m color[33m:[39m black [33m}[39m
 [90m   | [39m  [31m[1m^[22m[39m
 [90m 2 | [39m
"
`;

exports[`Errors Validation Error 1`] = `"options.sourceMap should be {String|Boolean} (https://github.com/postcss/postcss-loader#sourcemap)"`;
